actors/evm/src/lib.rs

```diff
@@ -173,6 +176,48 @@ impl EvmContractActor {
         let output = RawBytes::from(exec_status.output_data.to_vec());
         Ok(output)
     }
+
+    pub fn bytecode<BS, RT>(rt: &mut RT) -> Result<Cid, ActorError>
+    where
+        BS: Blockstore + Clone,
+        RT: Runtime<BS>,
+    {
+        // Any caller can fetch the bytecode of a contract; this is now EXT* opcodes work.
+        rt.validate_immediate_caller_accept_any()?;
+
+        let state: State = rt.state()?;
+        Ok(state.bytecode)
+    }
+
+    pub fn storage_at<BS, RT>(rt: &mut RT, params: GetStorageAtParams) -> Result<U256, ActorError>
+    where
+        BS: Blockstore + Clone,
+        RT: Runtime<BS>,
+    {
+        // This method cannot be called on-chain; other on-chain logic should not be able to
+        // access arbitrary storage keys from a contract.
+        rt.validate_immediate_caller_is([&fvm_shared::address::Address::new_id(0)])?;
```

Comment:
This very funky, we have an effectively uncallable method.

Thats one extra reason to expect the client to directly read the state, or it should be a non method entry point.

Having said that, it is totally fine for now, but we should think about this general pattern.